lib/Fmt_ast.ml

```diff
@@ -1183,9 +1179,7 @@ and fmt_pattern ?ext c ?pro ?parens ?(box = false)
   @@ fun c ->
   let parens = match parens with Some b -> b | None -> parenze_pat xpat in
   (match ctx0 with Pat {ppat_desc= Ppat_tuple _; _} -> hvbox 0 | _ -> Fn.id)
-  @@ ( match ppat_desc with
-     | Ppat_or _ -> fun k -> Cmts.fmt c ppat_loc @@ k
```

Comment:
Do we know why this was originally written? It looks like it's special-cased for *some* reason...

Reply:
The three diffs in this file only make sense together. We can remove this special case (to not print `pro` before the box in a `Ppat_or`) because later we remove the place where `Ppat_or` distinguishes itself from other types of patterns by printing its `pro` inside the box. Interchanging the order of the boxes in this way produces a pretty big diff, but it's mostly reversed by changing the indentation of the inner box when `nested = true` (which answers your comment below). I say "mostly" because there are some edge cases where the formatting _does_ still change --- namely, the strange cases that this PR fixes.

Why was `Ppat_or` written this way to begin with? I don't know. I am assuming the orginial author either didn't know or wasn't too worried about the cases where we've observed the original implementation produce bad behavior. I haven't noticed any regressions in the in-repo tests or when testing on the codebase, so I'm pretty confident this change is an improvement.